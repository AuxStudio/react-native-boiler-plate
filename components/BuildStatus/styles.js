import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
});

export default styles;
