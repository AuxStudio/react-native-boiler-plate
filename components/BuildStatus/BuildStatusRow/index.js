import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

import styles from './styles';

const propTypes = {
  labelText: PropTypes.string,
  valueText: PropTypes.string,
};

const defaultProps = {};

const BuildStatusRow = ({ labelText, valueText }) => {
  return (
    <Text style={styles.text}>
      <Text style={styles.boldText}>{labelText} </Text>
      {valueText}
    </Text>
  );
};

BuildStatusRow.propTypes = propTypes;
BuildStatusRow.defaultProps = defaultProps;

export default BuildStatusRow;
