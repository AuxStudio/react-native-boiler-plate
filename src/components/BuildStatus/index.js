import React from 'react';
import { View } from 'react-native';

import config from '../../config';
import styles from './styles';

import BuildStatusRow from './BuildStatusRow';

const propTypes = {};

const defaultProps = {};

const BuildStatus = () => {
  return (
    <View style={styles.container}>
      <BuildStatusRow
        labelText="Version:"
        valueText={`${config.version.major}.${config.version.minor}.${config.version.patch} | `}
      />

      <BuildStatusRow labelText="Build:" valueText={`${config.version.build} | `} />

      <BuildStatusRow labelText="Code:" valueText={`${config.version.code}`} />
    </View>
  );
};

BuildStatus.propTypes = propTypes;
BuildStatus.defaultProps = defaultProps;

export default BuildStatus;
