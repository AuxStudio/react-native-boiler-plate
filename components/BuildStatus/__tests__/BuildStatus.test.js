import React from 'react';
import renderer from 'react-test-renderer';

import BuildStatus from '..';

describe('BuildStatus', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<BuildStatus />);

      expect(component).toMatchSnapshot();
    });
  });
});
