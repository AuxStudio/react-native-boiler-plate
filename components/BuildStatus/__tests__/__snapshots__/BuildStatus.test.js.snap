// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildStatus renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}
    >
      Version:
       
    </Text>
    0.0.1 | 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}
    >
      Build:
       
    </Text>
    1 | 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}
    >
      Code:
       
    </Text>
    1
  </Text>
</View>
`;
